verilog/analysis/checkers/always_ff_non_blocking_rule_test.cc

```diff
@@ -0,0 +1,51 @@
+// Copyright 2017-2019 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "verilog/analysis/checkers/always_ff_non_blocking_rule.h"
+
+#include <initializer_list>
+
+#include "gtest/gtest.h"
+#include "common/analysis/linter_test_utils.h"
+#include "common/analysis/syntax_tree_linter_test_utils.h"
+#include "common/text/symbol.h"
+#include "verilog/CST/verilog_nonterminals.h"
+#include "verilog/CST/verilog_treebuilder_utils.h"
+#include "verilog/analysis/verilog_analyzer.h"
+
+namespace verilog {
+namespace analysis {
+namespace {
+
+using verible::LintTestCase;
+using verible::RunLintTestCases;
+
+TEST(AlwaysFFNonBlockingRule, FunctionFailures) {
+  constexpr int kToken = TK_always_ff;
+  const std::initializer_list<LintTestCase> kAlwaysFFNonBlockingTestCases = {
+      {""},
+      {"module m;\nendmodule\n"},
+      {"module m;\ninitial begin end\nendmodule"},
+      {"module m;\n", {kToken, "always_ff"}, " @(*) a = b;\nendmodule"},
+      {"module m;\n", {kToken, "always_ff"}, " @(*) begin a = b; end\nendmodule"},
+      {"module m;\nalways_comb a = b;\nendmodule"},
+      {"module m;\nalways_comb begin a = b; end\nendmodule"},
+  };
```

Comment:
Same suggested test cases from the other rule's test.

Reply:
Done.